--- nss_db-2.5/libnss_db.map~	2011-12-12 17:48:06.000000000 +0200
+++ nss_db-2.5/libnss_db.map	2012-02-09 23:09:44.849891839 +0200
@@ -15,6 +15,8 @@
     _nss_db_setnetgrent; _nss_db_setprotoent; _nss_db_setpwent;
     _nss_db_setrpcent; _nss_db_setservent; _nss_db_setspent;
     _nss_db_initgroups_dyn;
-  local:
-    *;
+};
+NSSDB_PRIVATE {
+  global:
+    *_nssdb;
 };
--- nss_db-2.5/Makefile.am~	2012-02-09 23:08:49.000000000 +0200
+++ nss_db-2.5/Makefile.am	2012-02-09 23:11:00.986686451 +0200
@@ -51,7 +51,7 @@
 
 bin_PROGRAMS = makedb
 makedb_SOURCES = makedb.c
-makedb_LDADD = @DB_LIBS@ @SELINUX_LIBS@  db-compat-copy-makedb.lo lib/libgnu.a
+makedb_LDADD = @DB_LIBS@ @SELINUX_LIBS@ libnss_db.la db-compat-copy-makedb.lo lib/libgnu.a
 makedb_CPPFLAGS = -I$(top_builddir)/lib -I$(top_srcdir)/lib
 db-compat-copy-makedb.c: db-compat.c
 	rm -f $@
