--- nss_db-2.2.3pre1.org/db-open.c	2004-08-16 06:20:08.219189152 +0200
+++ nss_db-2.2.3pre1/db-open.c	2004-08-16 06:20:32.966427000 +0200
@@ -58,7 +58,7 @@
   if (err != 0)
     {
       if (err > 0)
-	__set_errno (err);
+	errno = err;
       return NSS_STATUS_UNAVAIL;
     }
 
@@ -75,7 +75,7 @@
  fail:
   db->close (db, 0);
   if (err > 0)
-    __set_errno (err);
+    errno = err;
   return NSS_STATUS_UNAVAIL;
 }
 
