--- nss_db-2.5/db-Makefile.in~	2011-12-12 12:25:35.000000000 +0200
+++ nss_db-2.5/db-Makefile.in	2012-02-09 22:41:12.395398440 +0200
@@ -61,6 +61,7 @@
 		 /^[^#]/ { printf ".%s ", $$1; print; \
 			   printf "=%s ", $$3; print }' $^ | \
 	$(MAKEDB) -o $@ -
+	chmod 644 $@
 
 $(GROUP_DB): $(GROUP_SOURCE)
 	@echo "$@ ..."
@@ -71,6 +72,7 @@
 		 /^[^#]/ { printf ".%s ", $$1; print; \
 			   printf "=%s ", $$3; print }' $^ | \
 	$(MAKEDB) -o $@ -
+	chmod 644 $@
 
 $(ETHERS_DB): $(ETHERS_SOURCE)
 	@echo "$@ ..."
@@ -81,6 +83,7 @@
 		 /^[^#]/ { printf ".%s ", $$2; print; \
 			   printf "=%s ", $$1; print }' $^ | \
 	$(MAKEDB) -o $@ -
+	chmod 644 $@
 
 $(PROTO_DB): $(PROTO_SOURCE)
 	@echo "$@ ..."
@@ -93,6 +96,7 @@
 			   for (i = 3; i <= NF && !($$i ~ /^#/); ++i) \
 			     { printf ".%s ", $$i; print } }' $^ | \
 	$(MAKEDB) -o $@ -
+	chmod 644 $@
 
 $(RPC_DB): $(RPC_SOURCE)
 	@echo "$@ ..."
@@ -105,6 +109,7 @@
 			   for (i = 3; i <= NF && !($$i ~ /^#/); ++i) \
 			     { printf ".%s ", $$i; print } }' $^ | \
 	$(MAKEDB) -o $@ -
+	chmod 644 $@
 
 $(SERVICES_DB): $(SERVICES_SOURCE)
 	@echo "$@ ..."
@@ -120,6 +125,7 @@
 			     { printf ".%s/%s ", $$i, $$3; print; \
 			       printf ".%s/ ", $$i; print } }' $^ | \
 	$(MAKEDB) -o $@ -
+	chmod 644 $@
 
 $(SHADOW_DB): $(SHADOW_SOURCE)
 	@echo "$@ ..."
@@ -151,3 +157,4 @@
 			   gsub(/[ \t]+/, " "); \
 			   if(end == 1) printf "%s", $$0; else print }' $^ | \
 	$(MAKEDB) -o $@ -
+	chmod 644 $@
