--- nss_db-2.5/Makefile.am~	2012-02-09 22:59:18.000000000 +0200
+++ nss_db-2.5/Makefile.am	2012-02-09 23:00:19.130840181 +0200
@@ -45,7 +45,8 @@
 else
   libnss_db_la_SOURCES += db-alias.c
   EXTRA_libnss_db_la_SOURCES += libnss_db.map
-  libnss_db_la_LDFLAGS += -Wl,--version-script=libnss_db.map
+  libnss_db_la_LDFLAGS += -Wl,--version-script=libnss_db.map \
+	-Wl,$(slibdir)/libnss_files.so.2
 endif
 
 bin_PROGRAMS = makedb
