--- nss_db-2.2.3pre1/db-compat.c	2010-02-22 19:32:31.371616603 +0200
+++ nss_db-2.2.3pre1/db-compat.c	2010-02-22 19:32:31.371616603 +0200
@@ -39,7 +39,11 @@
   if (err)
     return err;
 
-  err = db->open (db, file, NULL, type, flags, mode);
+#if (DB_VERSION_MAJOR > 4) || ((DB_VERSION_MAJOR == 4) && (DB_VERSION_MINOR >= 1))
+  err = (db->open) (db, NULL, file, NULL, type, flags, mode);
+#else
+  err = (db->open) (db, file, NULL, type, flags, mode);
+#endif
   if (err)
     {
       db->close (db, 0);
